### PR TITLE
Fix job name generation

### PR DESCRIPTION
**Pull request by maronuu**
## What
Fix a way to generate job name.
- Validate a job name more precisely following the [k8s job name restriction](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set)
- Improve readability of job names
- There is almost no conflicts

### CODE REVIEW

File: kannon/master.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def _exec_bullet_task(self, task: TaskOnBullet) -> None:
         pkl_path = self._gen_pkl_path(task)
         make_target(pkl_path).dump(task)
         # Run on child job
-        job_name = gen_job_name(f"{self.job_prefix}-{task.get_task_family()}")
```
Comment (kitagry) [inline kannon/master.py:121]: [should] use `job_prefix` or delete `job_prefix` argument of `Kannon.__init__` . I think to use `job_prefix` is better.

**Review by kitagry — CHANGES_REQUESTED** [review]
LGTM except for comment

File: kannon/kube_util.py [hunk 2]
```diff
@@ -31,8 +31,9 @@ def get_job_status(api_instance: client.BatchV1Api, job_name: str, namespace: st
 
 
 def gen_job_name(job_prefix: str) -> str:
-    job_name = f"{job_prefix}-{str(random.randint(0, 255)).zfill(3)}-{datetime.now().strftime('%Y%m%d%H%M%S')}"
-    # TODO: validate job_name more precisely
-    job_name = job_name[:50]
+    job_suffix = f"{datetime.now().strftime('%Y%m%d%H%M%S')}-{str(random.randint(0, 255)).zfill(3)}"
+    job_prefix = job_prefix[:62 - len(job_suffix)]
```
Comment (yokomotod) [inline kannon/kube_util.py:35]: can you make constant variable instead of magic literal `62`, `63` ?

**Review by yokomotod — APPROVED** [review]
one comment in addition to kitagry's comment.

but basically lgtm.

File: kannon/master.py (hunk 1 above)
Comment (maronuu) [reply kannon/master.py:121]: Thank you, 
Added a change at commit 6a93fd2.

**Review by kitagry — APPROVED** [review]
LGTM

**Pull request merged**